File: DSA/upama_kumari_590014909/day_17_question_1.py
def binary_search(arr, target):
    left = 0
    right = len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2 

        if arr[mid] == target:
            return mid 
        elif arr[mid] < target:
            left = mid + 1  
        else:
            right = mid - 1 

    return -1 

array = [10, 20, 30, 40, 50]
target = 30
result = binary_search(array, target)
print(result)

File: DSA/Ayush_Mishra_590014844/Day_17_Question_1.c
#include <stdio.h>

int binarySearch(int arr[], int size, int target) {
    int left = 0, right = size - 1;
    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target) return mid;
        else if (arr[mid] < target) left = mid + 1;
        else right = mid - 1;
    }
    return -1;
}

int main() {
    int n, target;
    printf("Enter number of elements: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter sorted elements:\n");
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);
    printf("Enter target element: ");
    scanf("%d", &target);
    int result = binarySearch(arr, n, target);
    printf("Index of target: %d\n", result);
    return 0;
}
